fastparquet/util.py

```diff
@@ -276,18 +267,71 @@ def get_file_scheme(paths, sep='/'):
         return 'simple'
     if None in paths:
         return 'other'
-    parts = [p.split(sep) for p in paths]
+    parts = [p.split('/') for p in paths]
     lens = [len(p) for p in parts]
     if len(set(lens)) > 1:
         return 'other'
     if set(lens) == {1}:
         return 'flat'
-    s = ex_from_sep(sep)
-    dirs = [p.rsplit(sep, 1)[0] for p in paths]
+    s = ex_from_sep('/')
+    dirs = [p.rsplit('/', 1)[0] for p in paths]
     matches = [s.findall(d) for d in dirs]
     if all(len(m) == (l - 1) for (m, l) in
            zip(matches, lens)):
         keys = (tuple(m[0] for m in parts) for parts in matches)
         if len(set(keys)) == 1:
             return 'hive'
     return 'drill'
+
+
+def join_path(*path):
+    def scrub(i, p):
+        p = p.replace(os.sep, "/")
```

Comment:
So *this* is the windows magic? Would appreciate a comment.

Reply:
Done.